### PR TITLE
move common params from hostgroup to HostMixin

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
I've not added tests to host yet

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -175,10 +175,33 @@ def __init__(self, foreman_spec=None, **kwargs):
...
+            openscap_proxy=dict(type='entity', resource_type='smart_proxies'),
+            content_source=dict(type='entity', scope='organization', resource_type='smart_proxies'),
+            lifecycle_environment=dict(type='entity', scope='organization'),
+            kickstart_repository=dict(type='entity', scope='organization', resource_type='repositories'),
+            content_view=dict(type='entity', scope='organization'),
         )
         if foreman_spec:
             args.update(foreman_spec)
-        super(HostMixin, self).__init__(foreman_spec=args, **kwargs)
+        required_plugins = kwargs.pop('required_plugins', []) + [
```
Comment (mdellweg) [inline plugins/module_utils/foreman_helper.py]: It works for Mixins? :roller_coaster:

**Comment by evgeni** [comment]
Yes, just not for every param

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: yes

**Comment by evgeni** [comment]
added tests and dropped the "needs HG" requirement as that's not true anymore

File: tests/test_playbooks/vars/host.yml [hunk 2]
```diff
@@ -5,3 +5,9 @@ host:
   hostgroup: "test_group"
   domain: "example.net"
   subnet: Test subnet4
+  arch: superARCH
```
Comment (mdellweg) [inline tests/test_playbooks/vars/host.yml]: Is this a 42 bit architecture?

Comment (evgeni) [reply tests/test_playbooks/vars/host.yml]: 192!

It's a reference to https://en.wikipedia.org/wiki/SuperH ;-)

**Pull request merged**